### PR TITLE
fixed cosmos hashing

### PR DESCRIPTION
**Pull request by gdsoumya**
Signed-off-by: Soumya Ghosh Dastidar <gdsoumya@gmail.com>

### CODE REVIEW

File: multichain_test.go [hunk 1]
```diff
@@ -769,7 +769,7 @@ var _ = Describe("Multichain", func() {
 			},
 			{
 				senderEnv: func() (id.PrivKey, *id.PubKey, multichain.Address) {
-					pkEnv := os.Getenv("TERRA_PK")
+					pkEnv := "a96e62ed3955e65be32703f12d87b6b5cf26039ecfa948dc5107a495418e5330" //os.Getenv("TERRA_PK")
```
Comment (roynalnaruto) [inline multichain_test.go:772]: why this?

File: chain/cosmos/tx.go [hunk 2]
```diff
@@ -76,6 +79,13 @@ func (builder txBuilder) WithSignMode(signMode int32) txBuilder {
 // This transaction is unsigned, and must be signed before submitting to the
 // cosmos chain.
 func (builder txBuilder) BuildTx(ctx context.Context, from, to address.Address, value, nonce, gasLimit, gasPrice, gasCap pack.U256, payload pack.Bytes) (account.Tx, error) {
+	pubKeyBytes, err := hex.DecodeString(string(from))
```
Comment (jazg) [inline chain/cosmos/tx.go:82]: Probably worth leaving a comment here like:
```suggestion
	// We assume the "from" address is a public key as it is required for
	// setting the signature.
	pubKeyBytes, err := hex.DecodeString(string(from))
```

File: chain/cosmos/tx.go [hunk 3]
```diff
@@ -3,9 +3,14 @@ package cosmos
...
 	"fmt"
 	"math/big"
 
+	"github.com/renproject/multichain"
+
+	"github.com/cosmos/cosmos-sdk/x/auth/tx"
+
 	"github.com/btcsuite/btcd/btcec"
 
 	"github.com/cosmos/cosmos-sdk/client"
```
Comment (jazg) [inline chain/cosmos/tx.go:16]: Possible to format these? i.e. remove the spaces between them

File: multichain_test.go [hunk 4]
```diff
@@ -18,6 +18,8 @@ import (
 	"testing/quick"
 	"time"
 
+	"github.com/renproject/multichain/api/account"
+
```
Comment (jazg) [inline multichain_test.go:22]:
```suggestion
	"github.com/renproject/multichain/api/account"
```

Comment (jazg) [inline multichain_test.go:22]: Formatting

File: multichain_test.go [hunk 5]
```diff
@@ -931,6 +933,15 @@ var _ = Describe("Multichain", func() {
 							amount, nonce, gasLimit, gasPrice, gasCap,
 							payload,
 						)
+						if accountChain.chain == multichain.Terra {
```
Comment (jazg) [inline multichain_test.go:936]:
```suggestion
						// TODO: Temporarily use the public key in place of the
						// from address for Terra.
						if accountChain.chain == multichain.Terra {
```

File: chain/cosmos/tx.go (hunk 3 above)
Comment (gdsoumya) [reply chain/cosmos/tx.go:16]: I did goimport and gofmt not sure why it didn't change this one. Will try to manually change it

Comment (jazg) [reply chain/cosmos/tx.go:16]: I think they need to be grouped together for the auto formatting to apply

**Review by jazg — APPROVED** [review]
LGTM!

**Review by roynalnaruto — APPROVED** [review]

**Pull request merged**